Formula/python.rb

```diff
@@ -98,17 +98,25 @@ def install
     ldflags  = []
     cppflags = []
 
-    unless MacOS::CLT.installed?
-      # Help Python's build system (setuptools/pip) to build things on Xcode-only systems
+    if !MacOS::CLT.installed? || MacOS.version >= :mojave
+      # Help Python's build system (setuptools/pip) to build things on
+      # Xcode-only systems or systems where headers are not in /usr/include
+      if !MacOS::CLT.installed?
+        sdk = MacOS.sdk_path
+      else
+        sdk = "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk"
+      end
+
       # The setup.py looks at "-isysroot" to get the sysroot (and not at --sysroot)
-      cflags   << "-isysroot #{MacOS.sdk_path}"
-      ldflags  << "-isysroot #{MacOS.sdk_path}"
-      cppflags << "-I#{MacOS.sdk_path}/usr/include" # find zlib
+      cflags   << "-isysroot #{sdk}"
+      ldflags  << "-isysroot #{sdk}"
+      cppflags << "-I#{sdk}/usr/include" # find zlib
```

Comment:
I think if you remove the SDK from here, it will Just Work. `setup.py` uses `-isysroot` to get the sysroot, then concatenates paths itself. That means setting `-I/usr/include` sets a path it then uses to look inside the SDK and locate zlib's headers.